accepted/2024/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,193 @@
+2. The group to be published no longer has any packages in it. In this case the command should fail detailing the reason for the failure.
+3. This will be the same but for deprecation. Essentially, any bulk operation using groups, where the set of the packages in the group
+ have changed after staging, should result in an error with details about the change. The `--force` flag should again be used
+ to ignore the change and move forward with execution of the request (either publishing or deprecation). To summarize,
+ the fundamental principle is that a user cannot deprecate a package they don't own.
+
+#### Lifespan of a staged group
+If users keep staging packages but do nothing with them, NuGet will become an ever-growing package graveyard,
+ incurring higher and higher costs over time simply for storing these packages. To avoid this problem, stages should be time-limited.
+ That is, there should be a maximum lifespan for a stage. Any updates to the stage will reset its TTL date to some amount (let's say 30 days). If within the next 30 days no changes are made to the stage, NuGet should then remove the stage (all its packages basically).
+
+#### Expanding groups
+It should be possible for a group to expand over time. This is currently a real scenario, where .NET GA SDK ships a set of packages,
+ and then follow-up SDK updates ship updated packages which still belong to the same .NET 8 package group.
+ Here how this will play out with the tooling support described above:
+1. During GA release, the release team will stage the set of RTM packages for the release. (nuget stage --group-id "net8.0" --package-id <package-file-path>)
+2. Then on the day of the release, all the staged packages will be published using `nuget push --group "net8.0"` command.
+   At this point, the group with id `net8.0` will be empty.
+3. Later, as new builds are being prepared for a patch release, a new set of packages will be staged, to be later published to the same group.
+As there can be multiple candidate builds for a release, there is a need for functionality to discard a stage for a given group `nuget delete --stage "net8.0"`.
+This will remove all the staged but not published packages from the net8.0 group.
+4. A new set of packages will be staged for the 8.0 group, until the final build is known. At this point, there is only one unified set of related packages is staged.
+5. On the release date, the release team will call `nuget push --group "net8.0"` again, and only the staged packages will be published, resulting in an expanded set of published packages in the net8.0 group.
+6. Some time in the future, when the release is already out of support, somebody from the .NET team will call
+ `nuget deprecate --stage "net8.0"` and all the packages which have ever been published to that group will be deprecated.
+ Those packages, which has already been deprecated because of whatever reason, will not be altered.
+
+#### Asynchronous behavior
+
```

Comment:
NuGet.org supports 2 versions of the protocol: V2 and V3. Is the expectation both would become available roughly at the same time once group is published?

Reply:
To expand on that a bit: in current implementation V3 has its own publishing pipeline that is triggered strictly after package becomes available in V2. I doubt we can stage anything to V3 unless we rewrite the pipeline with that in mind.

Reply:
I think this one is an implementation / design problem that is better solved during development. Whether it will require some redesign or some clever new solution, I'm not sure about, as I'm not familiar with the internals.